### ISSUE TITLE
Do you have plans to support this plugin in Rider?

### ISSUE THREAD

**Issue by afucher** [dkandalov/limited-wip#35]
Hi,

I was using this plugin when I was working with Java, but now I'm working with C# and using Rider.
The plugin is available to download and install, but not work as expected. The TCR is not working :/

I would love to have this working on Rider, do you have checked if is something viable to do this?

**Comment by dkandalov** [comment]
Hi,

Thanks a lot for reporting this!

Looks like a bug in the plugin which should be fixed. Yes, I think it's worth fixing because in theory the plugin should work in IntelliJ IDEs.

**Comment by afucher** [comment]
Nice, it seems also that the line change counter is also wrong.

I don't have knowledge about developing plugins, but if I have some time I'll try to take a look. I've started to make C# live coding on twitch, and I want to use TCR, and this plugin will help me a lot!
Maybe I tried to make a live to learn and try to fix this bug, is an idea that occurs me now.

**Comment by afucher** [comment]
@dkandalov can you give me some steps about how can I run this project to debug? I tried to import on IntelliJ but was not able to configure to run and debug :/

**Comment by afucher** [comment]
Just have discovered the `intellij->runIde` task, now I'll try to discover how to run on Rider

**Comment by afucher** [comment]
Just to update, I was able to debug on Rider! I'll probably start to check this issue tomorrow, in a live code session on Twitch. But my live code is in Portuguese.
Hope I'll be able to identify and help with this issue

**Comment by afucher** [comment]
Hi @dkandalov , today I spent some time debugging the issue and I found a different behavior between IntelliJ and Rider.
IntellJ send a Notification when the Test Runner finishes with something like this `10:52 PM	Tests passed: 1`. But Rider doesn't send this Notification.

Making some research on the internet I found that is an abstract [TestStatusListener](https://github.com/JetBrains/intellij-community/blob/master/platform/testRunner/src/com/intellij/execution/testframework/TestStatusListener.java) that could be used to implment this.
But probably will need to make a lot of changes into this plugin.

I didn't tried to use this approach, I'll like to get your opinion related to this before try to use this Listener.

**Comment by dkandalov** [comment]
Oh, this explains why TCR doesn't work.

If `TestStatusListener` works and has all the information, it's good enough (I vaguely remember not using it for some reason, but if it works now this is great).

**Comment by afucher** [comment]
I need to check if it works, I don't know if it has all the information. The documentation of this is not so good as I was expecting.
I'll make some tests, and update this issue ASAP.

**Comment by dkandalov** [comment]
Yes, there is not much documentation but given that `TestStatusListener` is not part of "openapi" this is fair enough I guess. I usually check out IJ source code and browse it as documentation of how things are actually used.

I quickly looked at `TestStatusListener` and there are couple potential problems:
 - `notifySuiteFinished(AbstractTestProxy root)` is not passing the current project, so you essentially don't know the project in which it was called. This function is deprecated by maybe it's still called from unit testing plugin 🤷‍♂️
- in `notifySuiteFinished(@Nullable AbstractTestProxy root, Project project)` the `root` is nullable so there is not guarantee that there will be an object describing test results

In practice, all of the above might work just fine for majority of languages and testing frameworks. But the only way to really know is to manually test I guess.

**Comment by dkandalov** [comment]
Actually, I suspect that Rider doesn't use `TestStatusListener` at all. There is a chance it's C# code from resharper running in a separate process and using IJ api for UI.
